subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1522,3 +1522,33 @@ int nrf_cloud_parse_cell_pos_response(const char *const buf,
 	cJSON_Delete(cell_pos_obj);
 	return ret;
 }
+
+bool nrf_cloud_detect_disconnection_request(const char *const buf) {
+	if (buf == NULL) {
+		return false;
+	}
+
+	/* Fast test to avoid parsing EVERY message with cJSON. */
+	if (strstr(buf, NRF_CLOUD_JSON_APPID_VAL_DEVICE) == NULL ||
+	    strstr(buf, NRF_CLOUD_JSON_MSG_TYPE_VAL_DISCONNECT) == NULL) {
+		return false;
+	}
+
+	/* If the quick test passes, use cJSON to get certainty */
+	bool ret = true;
+	cJSON *discon_request_obj = cJSON_Parse(buf);
+
+	/* Check for nRF Cloud disconnection request MQTT message */
+	if (!json_item_string_exists(discon_request_obj, NRF_CLOUD_JSON_MSG_TYPE_KEY,
+				     NRF_CLOUD_JSON_MSG_TYPE_VAL_DISCONNECT) ||
+	    !json_item_string_exists(discon_request_obj, NRF_CLOUD_JSON_APPID_KEY,
+				     NRF_CLOUD_JSON_APPID_VAL_DEVICE)) {
+		/* Not a disconnection request message */
+		ret = false;
+		goto cleanup;
```

Comment:
goto not needed since the label is just below this section

Reply:
Understood, I included it to maintain structural consistency with the rest of nrf_cloud_codec, but that was probably not necessary. Will be removed!